actor/src/main/scala/org/apache/pekko/japi/function/Function.scala

```diff
@@ -152,3 +152,10 @@ trait Creator[+T] extends Serializable {
   @throws(classOf[Exception])
   def create(): T
 }
+
+@nowarn("msg=@SerialVersionUID has no effect")
+@SerialVersionUID(1L)
+@FunctionalInterface
+trait IntFunction[T] extends Serializable {
+  def apply(value: Int): T
```

Comment:
? what's the poin of this?

Reply:
We already have a place where we define all of our functional interfaces, its in `actor/src/main/scala/org/apache/pekko/japi/function/Function.scala`. In your PR you created another implementation in Java which means we have functional interfaces defined in both Scala and Java for no good reason.

Lets keep all of the functional interfaces defined where they already are in `actor/src/main/scala/org/apache/pekko/japi/function/Function.scala` so its consistent.

Reply:
The reason is in https://github.com/apache/pekko/issues/2185, ok?
Even Akka-sdk is now Java-first.

Reply:
These are Java APIs, so shouldn't they be written in Java? Just because they were done in the past doesn't mean they should be done now.

Reply:
> The reason is in #2185, ok? Even Akka-sdk is now Java-first.

If you are going to make a decision like move everything to Java (or to Scala) for that matter, you need to discuss it with community first. You can't just sneak it into PR's because you think its a good idea.

Please discuss this with community first, also whether to throw an exception or not is a separate matter from whether functions can throw exceptions.